differs/rpm_diff.go

```diff
@@ -123,6 +124,33 @@ func rpmDataFromContainer(image pkgutil.Image) (map[string]util.PackageInfo, err
 
 		defer client.RemoveImage(imageName)
 		defer logrus.Infof("Removing image %s", imageName)
+	} else if image.IsCloud() {
+		// if it's a remote image we pull it unconditionally and avoid
+		// name/tag conflicts by pulling it via its sha256 digest
+		var buf bytes.Buffer
+		digest, err := image.GetRemoteDigest()
+		if err != nil {
+			return packages, err
+		}
+		imageName = image.GetName() + "@" + digest
+		logrus.Infof("Pulling remote image %s", imageName)
+		pullOpts := godocker.PullImageOptions{
+			Repository:   imageName,
+			OutputStream: &buf,
+		}
+		if err := client.PullImage(pullOpts, godocker.AuthConfiguration{}); err != nil {
+			return packages, fmt.Errorf("Error pulling remote image %s: %s", imageName, err)
+		}
+		// log PullImage() output in debug mode
+		if logrus.GetLevel() == logrus.DebugLevel {
+			// log each line separately for consistency
+			for _, out := range strings.Split(buf.String(), "\n") {
```

Comment:
does the output stream not already have newlines in it? just wondering why this is necessary.

Reply:
Yes, the output stream has newlines. The issue here is that logrus will *not* prepend the log prefix to a new line, so only logging `buf.String()` would look like:
```
[level][counter] first line
second line
third line...
```

I find that very confusing as it looks broken, so I usually split strings to print each array entry separately.